<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> publications | Jason Yang </title> <meta name="author" content="Jason Yang"> <meta name="description" content=""> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%E2%9A%9B%EF%B8%8F&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://jsunn-y.github.io/publications/"> <script src="/assets/js/theme.js?a5ca4084d3b81624bcfa01156dae2b8e"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>initTheme();</script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"> <span class="font-weight-bold">Jason</span> Yang </a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">about </a> </li> <li class="nav-item active"> <a class="nav-link" href="/publications/">publications <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/repositories/">repositories </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">cv </a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title">publications</h1> <p class="post-description"></p> </header> <article> <div class="publications"> <h2 class="bibliography">2025</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> </div> <div id="li_evaluation_2024" class="col-sm-8"> <div class="title">Evaluation of Machine Learning-Assisted Directed Evolution Across Diverse Combinatorial Landscapes</div> <div class="author"> Francesca-Zhoufan Li, <em>Jason Yang</em>, Kadina E Johnston, and <span class="more-authors" title="click to view 3 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '3 more authors' ? 'Emre Guersoy, Yisong Yue, Frances H Arnold' : '3 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">3 more authors</span> </div> <div class="periodical"> <em>Cell Systems</em>, Sep 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.sciencedirect.com/science/article/pii/S2405471225002200?dgcid=coauthor" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Various machine learning-assisted directed evolution (MLDE) strategies have been shown to identify highfitness protein variants more efficiently than typical wet-lab directed evolution approaches. However, limited understanding of the factors influencing MLDE performance across diverse proteins has hindered optimal strategy selection for wet-lab campaigns. To address this, we systematically analyzed multiple MLDE strategies, including active learning and focused training using six distinct zero-shot predictors, across 16 diverse protein fitness landscapes. By quantifying landscape navigability with six attributes, we found that MLDE offers a greater advantage on landscapes which are more challenging for directed evolution, especially when focused training is combined with active learning. Despite varying levels of advantage across landscapes, focused training with zero-shot predictors leveraging distinct evolutionary, structural, and stability knowledge sources consistently outperforms random sampling for both binding interactions and enzyme activities. Our findings provide practical guidelines for selecting MLDE strategies for protein engineering.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> </div> <div id="lambert_sequence-based_nodate" class="col-sm-8"> <div class="title">Sequence-Based Generative AI-Guided Design of Versatile Tryptophan Synthases</div> <div class="author"> Theophile Lambert, Amin Tavakoli, Gautham Dharuman, and <span class="more-authors" title="click to view 5 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '5 more authors' ? 'Jason Yang, Sukhvinder Kaur, Matthew Hill, Arvind Ramanathan, Frances H Arnold' : '5 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">5 more authors</span> </div> <div class="periodical"> <em>bioRxiv</em>, Aug 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.biorxiv.org/content/10.1101/2025.08.30.673177v1" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Enzymes offer unparalleled selectivity and sustainability for chemical synthesis, yet their widespread industrial application is often hindered by the slow and uncertain process of discovering and optimizing suitable biocatalysts. While directed evolution remains the gold standard for enzyme optimization, its success hinges on the availability of a starting enzyme with measurable activity, a persistent bottleneck for many desired functions. Designing libraries likely to contain such functional starting points remains a major challenge. In this work, we use the GenSLM protein language model (PLM) along with a series of filters to generate novel sequences of the β-subunit of tryptophan synthase (TrpB) that express in Escherichia coli, are stable, and are catalytically active in the absence of a TrpA partner. Many generated TrpBs also demonstrated significant substrate promiscuity, accepting non-canonical substrates typically inaccessible to natural TrpBs. Remarkably, several outperformed both natural and laboratory-optimized TrpBs on native and non-canonical substrates. Comparative analysis of the most active and promiscuous generated TrpB and its closest natural homolog confirmed that this enhanced functional versatility does not stem from the natural enzyme, highlighting the creative potential of generative models. Our results demonstrate that the model can generate enzymes which not only preserve natural structure and function but also acquire non-natural properties, establishing PLMs as powerful tools for biocatalyst discovery and engineering, with the potential in some cases to bypass further optimization.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/illuminating-480.webp 480w,/assets/img/publication_preview/illuminating-800.webp 800w,/assets/img/publication_preview/illuminating-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/illuminating.jpg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="illuminating.jpg" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="yang_illuminating_2025" class="col-sm-8"> <div class="title">Illuminating the universe of enzyme catalysis in the era of artificial intelligence</div> <div class="author"> <em>Jason Yang</em>, Francesca-Zhoufan Li, Yueming Long, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Frances H. Arnold' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>Cell Systems</em>, Aug 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.sciencedirect.com/science/article/pii/S2405471225002054" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Scientific research has revealed only a minuscule fraction of the enzymes that evolution has generated to power life’s essential chemical reactions—and an even tinier fraction of the vast universe of possible enzymes. Beyond the enzymes already annotated lie an astronomical number of biocatalysts that could enable sustainable chemical production, degrade toxic pollutants, and advance disease diagnosis and treatment. For the past few decades, directed evolution has been a powerful strategy for reshaping enzymes to access new chemical transformations: by harnessing nature’s existing diversity as a starting point and taking inspiration from nature’s most powerful design process, evolution, to modify enzymes incrementally. Recently, artificial intelligence (AI) methods have started revolutionizing how we understand and compose the language of life. In this perspective, we discuss a vision for AI-driven enzyme discovery to unveil a world of enzymes that transcends biological evolution and perhaps offers a route to genetically encoding almost any chemistry.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/SGPO-480.webp 480w,/assets/img/publication_preview/SGPO-800.webp 800w,/assets/img/publication_preview/SGPO-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/SGPO.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="SGPO.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="yang_steering_2025" class="col-sm-8"> <div class="title">Steering Generative Models with Experimental Data for Protein Fitness Optimization</div> <div class="author"> <em>Jason Yang</em>, Wenda Chu, Daniel Khalil, and <span class="more-authors" title="click to view 4 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '4 more authors' ? 'Raul Astudillo, Bruce J. Wittmann, Frances H. Arnold, Yisong Yue' : '4 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">4 more authors</span> </div> <div class="periodical"> <em>NeurIPS</em>, Dec 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://arxiv.org/abs/2505.15093v1" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="https://github.com/jsunn-y/SGPO" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="abstract hidden"> <p>Protein fitness optimization involves finding a protein sequence that maximizes desired quantitative properties in a combinatorially large design space of possible sequences. Recent developments in steering protein generative models (e.g diffusion models, language models) offer a promising approach. However, by and large, past studies have optimized surrogate rewards and/or utilized large amounts of labeled data for steering, making it unclear how well existing methods perform and compare to each other in real-world optimization campaigns where fitness is measured by low-throughput wet-lab assays. In this study, we explore fitness optimization using small amounts (hundreds) of labeled sequence-fitness pairs and comprehensively evaluate strategies such as classifier guidance and posterior sampling for guiding generation from different discrete diffusion models of protein sequences. We also demonstrate how guidance can be integrated into adaptive sequence selection akin to Thompson sampling in Bayesian optimization, showing that plug-and-play guidance strategies offer advantages compared to alternatives such as reinforcement learning with protein language models.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/alde-480.webp 480w,/assets/img/publication_preview/alde-800.webp 800w,/assets/img/publication_preview/alde-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/alde.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="alde.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="yang_active_2024" class="col-sm-8"> <div class="title">Active Learning-Assisted Directed Evolution</div> <div class="author"> <em>Jason Yang</em>, Ravi G Lal, James C Bowden, and <span class="more-authors" title="click to view 6 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '6 more authors' ? 'Raul Astudillo, Mikhail A Hameedi, Sukhvinder Kaur, Matthew Hill, Yisong Yue, Frances H Arnold' : '6 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">6 more authors</span> </div> <div class="periodical"> <em>Nature Communications</em>, Jan 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.nature.com/articles/s41467-025-55987-8" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="https://github.com/jsunn-y/ALDE" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="abstract hidden"> <p>Directed evolution (DE) is a powerful tool to optimize protein fitness for a specific application. However, DE can be inefficient when mutations exhibit non-additive, or epistatic, behavior. Here, we present Active Learning-assisted Directed Evolution (ALDE), an iterative machine learningassisted DE workflow that leverages uncertainty quantification to explore the search space of proteins more efficiently than current DE methods. We apply ALDE to an engineering landscape that is challenging for DE: optimization of five epistatic residues in the active site of an enzyme. In three rounds of wet-lab experimentation, we improve the yield of a desired product of a nonnative cyclopropanation reaction from 12% to 93%. We also perform computational simulations on existing protein sequence-fitness datasets to support our argument that ALDE can be more effective than DE. Overall, ALDE is a practical and broadly applicable strategy to unlock improved protein engineering outcomes.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2024</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/creep-480.webp 480w,/assets/img/publication_preview/creep-800.webp 800w,/assets/img/publication_preview/creep-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/creep.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="creep.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="yang_care_2024" class="col-sm-8"> <div class="title">CARE: a Benchmark Suite for the Classification and Retrieval of Enzymes</div> <div class="author"> <em>Jason Yang</em>, Ariane Mora, Shengchao Liu, and <span class="more-authors" title="click to view 4 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '4 more authors' ? 'Bruce J. Wittmann, Anima Anandkumar, Frances H. Arnold, Yisong Yue' : '4 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">4 more authors</span> </div> <div class="periodical"> <em>NeurIPS</em>, Dec 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="http://arxiv.org/abs/2406.15669" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="https://github.com/jsunn-y/CARE" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="abstract hidden"> <p>Enzymes are important proteins that catalyze chemical reactions. In recent years, machine learning methods have emerged to predict enzyme function from sequence; however, there are no standardized benchmarks to evaluate these methods. We introduce CARE, a benchmark and dataset suite for the Classification And Retrieval of Enzymes (CARE). CARE centers on two tasks: (1) classification of a protein sequence by its enzyme commission (EC) number and (2) retrieval of an EC number given a chemical reaction. For each task, we design train-test splits to evaluate different kinds of out-of-distribution generalization that are relevant to real use cases. For the classification task, we provide baselines for state-of-the-art methods. Because the retrieval task has not been previously formalized, we propose a method called Contrastive Reaction-EnzymE Pretraining (CREEP) as one of the first baselines for this task and compare it to the recent method, CLIPZyme. CARE is available at https://github.com/jsunn-y/CARE/.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/procalm-480.webp 480w,/assets/img/publication_preview/procalm-800.webp 800w,/assets/img/publication_preview/procalm-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/procalm.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="procalm.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="yang_conditional_2024" class="col-sm-8"> <div class="title">Function-Guided Conditional Generation Using Protein Language Models with Adapters</div> <div class="author"> <em>Jason Yang</em>, Aadyot Bhatnagar, Jeffrey A. Ruffolo, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Ali Madani' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>MLSB Workshop</em>, Dec 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://arxiv.org/abs/2410.03634" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="https://github.com/jsunn-y/ProCALM" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="abstract hidden"> <p>The conditional generation of proteins with desired functions is a key goal for generative models. Existing methods based on prompting of protein language models (PLMs) can generate proteins conditioned on a target functionality, such as a desired enzyme family. However, these methods are limited to simple, tokenized conditioning and have not been shown to generalize to unseen functions. In this study, we propose ProCALM (Protein Conditionally Adapted Language Model), an approach for the conditional generation of proteins using adapters to PLMs. While previous methods have used adapters for structure-conditioned generation from PLMs, our implementation of ProCALM involves finetuning ProGen2 to condition generation based on versatile representations of protein function-e.g. enzyme family, taxonomy, or natural language descriptions. ProCALM matches or exceeds the performance of existing methods at conditional sequence generation from target functions. Impressively, it can also generalize to rare and unseen functions. Overall, ProCALM is a flexible and computationally efficient approach, and we expect that it can be extended to a wide range of generative language models.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/trpB-480.webp 480w,/assets/img/publication_preview/trpB-800.webp 800w,/assets/img/publication_preview/trpB-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/trpB.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="trpB.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="johnston_combinatorially_nodate" class="col-sm-8"> <div class="title">A combinatorially complete epistatic fitness landscape in an enzyme active site</div> <div class="author"> Kadina E Johnston, Patrick J Almhjell, Ella J Watkins-Dulaney, and <span class="more-authors" title="click to view 4 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '4 more authors' ? 'Grace Liu, Jason Yang, Nicholas J Porter, Frances H Arnold' : '4 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">4 more authors</span> </div> <div class="periodical"> <em>PNAS</em>, Jul 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.pnas.org/doi/10.1073/pnas.2400439121" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Protein engineering often targets binding pockets or active sites which are enriched in epistasis—non-additive interactions between amino acid substitutions—and where the combined effects of multiple single substitutions are difficult to predict. Few existing sequence-fitness datasets capture epistasis at large scale, especially for enzyme catalysis, limiting the development and assessment of model-guided enzyme engineering approaches. We present here a combinatorially complete, 160,000-variant fitness landscape across four residues in the active site of an enzyme. Assaying the native reaction of a thermostable β-subunit of tryptophan synthase (TrpB) in a non-native environment yielded a landscape characterized by significant epistasis and many local optima. These effects prevent simulated directed evolution approaches from efficiently reaching the global optimum. There is nonetheless wide variability in the effectiveness of different directed evolution approaches, which together provide experimental benchmarks for computational and machine learning workflows. The most-fit TrpB variants contain a substitution that is nearly absent in natural TrpB sequences—a result that conservationbased predictions would not capture. Thus, although fitness prediction using evolutionary data can enrich in more-active variants, these approaches struggle to identify and differentiate among the most-active variants, even for this near-native function. Overall, this work presents a new, large-scale testing ground for model-guided enzyme engineering and suggests that efficient navigation of epistatic fitness landscapes can be improved by advances in both machine learning and physical modeling.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/outlook-480.webp 480w,/assets/img/publication_preview/outlook-800.webp 800w,/assets/img/publication_preview/outlook-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/outlook.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="outlook.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="yang_opportunities_2024" class="col-sm-8"> <div class="title">Opportunities and Challenges for Machine Learning-Assisted Enzyme Engineering</div> <div class="author"> <em>Jason Yang</em>, Francesca-Zhoufan Li, and Frances H. Arnold </div> <div class="periodical"> <em>ACS Central Science</em>, Feb 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://pubs.acs.org/doi/10.1021/acscentsci.3c01275" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Enzymes can be engineered at the level of their amino acid sequences to optimize key properties such as expression, stability, substrate range, and catalytic efficiency�or even to unlock new catalytic activities not found in nature. Because the search space of possible proteins is vast, enzyme engineering usually involves discovering an enzyme starting point that has some level of the desired activity followed by directed evolution to improve its “fitness” for a desired application. Recently, machine learning (ML) has emerged as a powerful tool to complement this empirical process. ML models can contribute to (1) starting point discovery by functional annotation of known protein sequences or generating novel protein sequences with desired functions and (2) navigating protein fitness landscapes for fitness optimization by learning mappings between protein sequences and their associated fitness values. In this Outlook, we explain how ML complements enzyme engineering and discuss its future potential to unlock improved engineering outcomes.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2023</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/decoil-480.webp 480w,/assets/img/publication_preview/decoil-800.webp 800w,/assets/img/publication_preview/decoil-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/decoil.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="decoil.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="yang_decoil_2023" class="col-sm-8"> <div class="title">DeCOIL: Optimization of Degenerate Codon Libraries for Machine Learning-Assisted Protein Engineering</div> <div class="author"> <em>Jason Yang</em>, Julie Ducharme, Kadina E. Johnston, and <span class="more-authors" title="click to view 3 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '3 more authors' ? 'Francesca-Zhoufan Li, Yisong Yue, Frances H. Arnold' : '3 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">3 more authors</span> </div> <div class="periodical"> <em>ACS Synthetic Biology</em>, Jul 2023 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://pubs.acs.org/doi/10.1021/acssynbio.3c00301" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="https://github.com/jsunn-y/DeCOIL" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="abstract hidden"> <p>With advances in machine learning (ML)-assisted protein engineering, models based on data, biophysics, and natural evolution are being used to propose informed libraries of protein variants to explore. Synthesizing these libraries for experimental screens is a major bottleneck, as the cost of obtaining large numbers of exact gene sequences is often prohibitive. Degenerate codon (DC) libraries are a cost-effective alternative for generating combinatorial mutagenesis libraries where mutations are targeted to a handful of amino acid sites. However, existing computational methods to optimize DC libraries to include desired protein variants are not well suited to design libraries for ML-assisted protein engineering. To address these drawbacks, we present DEgenerate Codon Optimization for Informed Libraries (DeCOIL), a generalized method that directly optimizes DC libraries to be useful for protein engineering: to sample protein variants that are likely to have both high fitness and high diversity in the sequence search space. Using computational simulations and wet-lab experiments, we demonstrate that DeCOIL is effective across two specific case studies, with the potential to be applied to many other use cases. DeCOIL offers several advantages over existing methods, as it is direct, easy to use, generalizable, and scalable. With accompanying software (https://github.com/jsunn-y/DeCOIL), DeCOIL can be readily implemented to generate desired informed libraries.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2022</h2> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/polymers-480.webp 480w,/assets/img/publication_preview/polymers-800.webp 800w,/assets/img/publication_preview/polymers-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/polymers.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="polymers.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="yang_machine_2022" class="col-sm-8"> <div class="title">Machine learning enables interpretable discovery of innovative polymers for gas separation membranes</div> <div class="author"> <em>Jason Yang</em>, Lei Tao, Jinlong He, and <span class="more-authors" title="click to view 2 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '2 more authors' ? 'Jeffrey R. McCutcheon, Ying Li' : '2 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">2 more authors</span> </div> <div class="periodical"> <em>Science Advances</em>, Jul 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.science.org/doi/10.1126/sciadv.abn9545" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> <a href="https://github.com/jsunn-y/PolymerGasMembraneML" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">Code</a> </div> <div class="abstract hidden"> <p>Polymer membranes perform innumerable separations with far-reaching environmental implications. Despite decades of research, design of new membrane materials remains a largely Edisonian process. To address this shortcoming, we demonstrate a generalizable, accurate machine learning (ML) implementation for the discovery of innovative polymers with ideal performance. Specifically, multitask ML models are trained on experimental data to link polymer chemistry to gas permeabilities of He, H 2 , O 2 , N 2 , CO 2 , and CH 4 . We interpret the ML models and extract valuable insights into the contributions of different chemical moieties to permeability and selectivity. We then screen over 9 million hypothetical polymers and identify thousands that lie well above current performance upper bounds, including hundreds of never-before-seen ultrapermeable polymer membranes with O 2 and CO 2 permeability greater than 10 4 and 10 5 Barrers, respectively. High-fidelity molecular dynamics simulations confirm the ML-predicted gas permeabilities of the promising candidates, which suggests that many can be translated to reality. , Machine learning generates chemical insights for the design of polymeric gas separation membranes with exceptional performance.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> </div> <div id="he_molecular_2022" class="col-sm-8"> <div class="title">Molecular insights into the structure-property relationships of 3D printed polyamide reverse-osmosis membrane for desalination</div> <div class="author"> Jinlong He, <em>Jason Yang</em>, Jeffrey R. McCutcheon, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Ying Li' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>Journal of Membrane Science</em>, Sep 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.sciencedirect.com/science/article/pii/S0376738822004768" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>3D-printing is an emerging method for manufacturing polyamide (PA) reserve osmosis (RO) membranes for water treatment and desalination, which can precisely control membrane structural properties, such as thickness, roughness, and resolution. However, the synthesis-structure (i.e., degree of cross-linking (DC), m-phenylenediamine/trimesoyl chloride (MPD/TMC) ratio, and membrane thickness) to property (permeability and water-salt selectivity) relationships for these membranes has not been well understood. At the same time, a microscopic understanding of the physical mechanism of water and salt transport is needed to guide the design of high-performance 3D-printed membranes and improve the printing efficiency. Thus, the atomic-scale transport features and energetics of water and salt ions are studied at high pressure for the 3D-printed PA RO membranes with the different DCs and MPD/TMC ratios through non-equilibrium molecular dynamics (NEMD) simulations. Factoring in membrane structure properties, rejection ratio of salt ions and pressure-dependent water flux, 3D-printed PA membranes having an MPD/TMC ratio of 3.0:2.0 and a DC between 80%∼90% attains ideal performance: high water flux, high rejection of salt ions, and excellent structural integrity. Mechanistically, water permeability for highly cross-linked PA RO membranes depends on the temporary on-and-off channels that allow water molecules to jump from one cavity to another at high pressure. In addition, higher pressures cause rapid compaction of PA membranes’ free volume and membrane thickness. Membrane failure at high pressure is determined by the DC and MPD/TMC ratios-dependent compressive yield strength. In short, these findings provide physical insights for optimizing existing PA membranes and designing next-generation desalination membranes at the molecular level.</p> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> </div> <div id="duchanois_designing_2022" class="col-sm-8"> <div class="title">Designing polymeric membranes with coordination chemistry for high-precision ion separations</div> <div class="author"> Ryan M. DuChanois, Mohammad Heiranian, <em>Jason Yang</em>, and <span class="more-authors" title="click to view 5 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '5 more authors' ? 'Cassandra J. Porter, Qilin Li, Xuan Zhang, Rafael Verduzco, Menachem Elimelech' : '5 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">5 more authors</span> </div> <div class="periodical"> <em>Science Advances</em>, Mar 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.science.org/doi/10.1126/sciadv.abm9436" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>State-of-the-art polymeric membranes are unable to perform the high-precision ion separations needed for technologies essential to a circular economy and clean energy future. Coordinative interactions are a mechanism to increase sorption of a target species into a membrane, but the effects of these interactions on membrane permeability and selectivity are poorly understood. We use a multilayered polymer membrane to assess how ion-membrane binding energies affect membrane permeability of similarly sized cations: Cu 2+ , Ni 2+ , Zn 2+ , Co 2+ , and Mg 2+ . We report that metals with higher binding energy to iminodiacetate groups of the polymer more selectively permeate through the membrane in multisalt solutions than single-salt solutions. In contrast, weaker binding species are precluded from diffusing into the polymer membrane, which leads to passage proportional to binding energy and independent of membrane thickness. Our findings demonstrate that selectivity of polymeric membranes can markedly increase by tailoring ion-membrane binding energy and minimizing membrane thickness. , Tailoring interactions between a target ion and polymer can lead to highly precise ion separations for ultrathin membranes.</p> </div> </div> </div> </li> </ol> <h2 class="bibliography">2021</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/separation-480.webp 480w,/assets/img/publication_preview/separation-800.webp 800w,/assets/img/publication_preview/separation-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/separation.jpg" class="preview z-depth-1 rounded" width="100%" height="auto" alt="separation.jpg" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="yang_efficient_2021" class="col-sm-8"> <div class="title">Efficient separation of small organic contaminants in water using functionalized nanoporous graphene membranes: Insights from molecular dynamics simulations</div> <div class="author"> <em>Jason Yang</em>, Zhiqiang Shen, Jinlong He, and <span class="more-authors" title="click to view 1 more author" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '1 more author' ? 'Ying Li' : '1 more author'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">1 more author</span> </div> <div class="periodical"> <em>Journal of Membrane Science</em>, Jul 2021 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://www.sciencedirect.com/science/article/pii/S0376738821002805" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Small organic molecules, and specifically micropollutants, pose known hazards to human health, but their removal is particularly challenging in water treatment. Our work demonstrates promising single-layer nanoporous graphene (NPG) membranes with high water permeability and varying degrees of selectivity against common organic contaminants, using molecular dynamics (MD) simulations. Seven target organic molecules are considered—including methanol, urea, ethanol, 2-propanol, n-nitrosodimethylamine (NDMA), pyrrole, and phenol—to understand the molecular parameters that govern organic removal. We systematically study molecular transport dynamics and energetics through membranes having varying sizes of pores with hydrophobic (hydrogenated) and hydrophilic (hydroxylated) functionalizations. We find that NPG membranes with smaller, hydroxylated pores offer higher water/organic permselectivity compared to those with larger, hydrogenated pores, as they impede the transport of organics while facilitating the transport of water. Molecular size is the primary organic parameter that controls transport. Larger organic molecules have a greater affinity for the membrane and pore groups (interfacial-affinity sieving), but they are more hindered from entering the pore (pore-size sieving). There is a net decrease in transport with increased molecular size, which suggests that pore-size sieving is the governing mechanism for most of the molecular separations. We further find that traditional measures of organic hydrophobicity are not strong predictors of transport, as they correlate poorly with interfacial affinity, and the contribution of dehydration to the energy barrier of permeation is limited, though non-negligible. Finally, our simulations show that organic-pore interactions are not pressure independent: as flow rate is decreased, the effects of interfacial-affinity sieving become more pronounced. In short, this study establishes a comprehensive understanding of membrane and molecular parameters to guide the design of effective NPG membranes for organic contaminant removal, particularly in desalination and wastewater reuse.</p> </div> </div> </div> </li></ol> <h2 class="bibliography">2019</h2> <ol class="bibliography"><li> <div class="row"> <div class="col col-sm-2 abbr"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/publication_preview/vesicle-480.webp 480w,/assets/img/publication_preview/vesicle-800.webp 800w,/assets/img/publication_preview/vesicle-1400.webp 1400w," sizes="200px" type="image/webp"></source> <img src="/assets/img/publication_preview/vesicle.png" class="preview z-depth-1 rounded" width="100%" height="auto" alt="vesicle.png" data-zoomable="" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </picture> </figure> </div> <div id="zucker_shape-dependent_2019" class="col-sm-8"> <div class="title">Shape-Dependent Interactions of Manganese Oxide Nanomaterials with Lipid Bilayer Vesicles</div> <div class="author"> Ines Zucker, Sara M. Hashmi, <em>Jason Yang</em>, and <span class="more-authors" title="click to view 3 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '3 more authors' ? 'Yulian He, Lisa D. Pfefferle, Menachem Elimelech' : '3 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.text(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">3 more authors</span> </div> <div class="periodical"> <em>Langmuir</em>, Oct 2019 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1021/acs.langmuir.9b02428" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">HTML</a> </div> <div class="abstract hidden"> <p>Interactions of transition-metal-oxide nanomaterials with biological membranes have important environmental implications and applications in ecotoxicity and life-cycle assessment analysis. In this study, we quantitatively assess the impact of MnO2 nanomaterial morphology—one-dimensional (1D) nanowires, 2D nanosheets, and 3D nanoflowers—on their interaction with phospholipid vesicles as a model for biological membranes. Confocal microscopy suggests visual evidence for the interaction of undisrupted vesicles with dispersed MnO2 nanomaterials of different morphologies, and it further supports the observation that minimal dye leakage of the vesicle inner solution was detected during the interaction with MnO2 nanomaterials during the dye leakage assay. Upon titration of vesicles to dispersions of MnO2 nanowires, nanosheets, and nanoflowers, each roughly 10 times larger than the vesicles, dynamic light scattering reveals two diffusive time scales associated with aggregates in the mixture. While the longer time scale corresponds to the dispersed MnO2 control population, the appearance of a shorter timescale with vesicle addition indicates interaction between the dispersed metal oxide nanomaterials and the vesicles. The interaction is shape-dependent, being more pronounced for MnO2 nanowires than for nanosheets and nanoflowers. Furthermore, the shorter diffusive time scale is intermediate between the vesicle and nanomaterial controls, which may suggest a degree of metal oxide aggregate breakup. Vesicle adsorption isotherms and zeta potential measurements during titration corroborate vesicle attachment on the nanomaterials. Our results suggest that the dispersed nanomaterial shape plays an important role in mediating nondestructive vesicle–nanomaterial interactions and that lipid vesicles act as efficient surfactants for MnO2 nanomaterials.</p> </div> </div> </div> </li></ol> </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> © Copyright 2025 Jason Yang. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?b7816bd189846d29eded8745f9c4cf77"></script> <script defer src="/assets/js/copy_code.js?12775fdf7f95e901d7119054556e495f" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.min.js"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>